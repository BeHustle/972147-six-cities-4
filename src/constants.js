const CARD_TYPES = [
  `Apartment`,
  `Private room`
];

const Cities = {
  AMSTERDAM: [52.38333, 4.9]
};

const MAP_ZOOM = 12;

const ICON_SIZE = [30, 30];

const ICON_PATH = `img/pin.svg`;

export {CARD_TYPES, Cities, MAP_ZOOM, ICON_SIZE, ICON_PATH};
