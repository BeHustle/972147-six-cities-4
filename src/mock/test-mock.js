export const Mock = {
  countOffers: 128,
  userEmail: `Oliver.conner@gmail.com`,
  cardPrice: 256,
  cardName: `Beautiful & luxurious apartment at great location`,
  cardType: `Appartment`,
};
