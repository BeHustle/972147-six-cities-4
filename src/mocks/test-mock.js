export const Mock = {
  userEmail: `Oliver.conner@gmail.com`,
  cities: [{
    id: 1,
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
  }, {
    id: 2,
    name: `Paris`,
    coordinates: [52.38333, 4.9],
  }],
  offers: [{
    id: 1,
    name: `Beautiful & luxurious apartment at great locatccion`,
    price: 132,
    image: `img/apartment-01.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    type: `Hotel`,
    isPremium: true,
    inBookmarks: true,
    rating: 4.1,
    rooms: `3 Bedrooms`,
    guests: `Max 4 adults`,
    facilities: [`Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    author: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      isSuper: true,
    },
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    coordinates: [52.3909553943508, 4.85309666406198],
    cityId: 1,
  }],
};
