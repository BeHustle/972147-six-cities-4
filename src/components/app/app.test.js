import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {Sorts} from '../../constants.js';
import App from './app.jsx';
import {Mock} from '../../mocks/test-mock.js';

jest.mock(`../map/map.jsx`, () => `Map`);

const mockStore = configureStore([]);

it(`App component test`, () => {
  const store = mockStore({
    offers: Mock.offers,
    city: Mock.cities[0],
    cities: Mock.cities,
    userEmail: Mock.userEmail,
    sorts: Sorts,
    activeSort: Sorts.POPULAR,
    activeOfferId: null
  });

  const tree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
      {
        createNodeMock: () => {
          return document.createElement(`div`);
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
