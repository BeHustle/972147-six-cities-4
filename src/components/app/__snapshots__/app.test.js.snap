// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  user@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index 2"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Amsterdam
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item false"
              href="#"
              onClick={[Function]}
            >
              <span>
                Paris
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item false"
              href="#"
              onClick={[Function]}
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item false"
              href="#"
              onClick={[Function]}
            >
              <span>
                Brussels
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item false"
              href="#"
              onClick={[Function]}
            >
              <span>
                Hamburg
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item false"
              href="#"
              onClick={[Function]}
            >
              <span>
                Dusseldorf
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            2
             places to stay in 
            Amsterdam
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom false"
            >
              <li
                className="places__option places__option--active"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option false"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option false"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option false"
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="places__list cities__places-list tabs__content"
          >
            <article
              className="place-card cities__place-card"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/apartment-01.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      132
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button place-card__bookmark-button--active"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Beautiful & luxurious apartment at great locatccion
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  hotel
                </p>
              </div>
            </article>
            <article
              className="place-card cities__place-card"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/apartment-02.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      254
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button false"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Wood and stone place
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  apartment
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <map
            coordinates={
              Array [
                52.38333,
                4.9,
              ]
            }
            offers={
              Array [
                Object {
                  "author": Object {
                    "avatar": "img/avatar-angelina.jpg",
                    "isSuper": true,
                    "name": "Angelina",
                  },
                  "cityId": 1,
                  "coordinates": Array [
                    52.3909553943508,
                    4.85309666406198,
                  ],
                  "facilities": Array [
                    "Wi-Fi",
                    "Washing machine",
                    "Towels",
                    "Heating",
                    "Coffee machine",
                    "Baby seat",
                    "Kitchen",
                    "Dishwasher",
                    "Cabel TV",
                    "Fridge",
                  ],
                  "guests": "Max 4 adults",
                  "id": 1,
                  "image": "img/apartment-01.jpg",
                  "images": Array [
                    "img/room.jpg",
                    "img/apartment-01.jpg",
                    "img/apartment-02.jpg",
                    "img/apartment-03.jpg",
                    "img/studio-01.jpg",
                    "img/apartment-01.jpg",
                  ],
                  "inBookmarks": true,
                  "isPremium": true,
                  "name": "Beautiful & luxurious apartment at great locatccion",
                  "price": 132,
                  "rating": 4.1,
                  "rooms": "3 Bedrooms",
                  "text": Array [
                    "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.",
                    "An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.",
                  ],
                  "type": "hotel",
                  "zoom": 10,
                },
                Object {
                  "author": Object {
                    "avatar": "img/avatar-max.jpg",
                    "isSuper": true,
                    "name": "Max",
                  },
                  "cityId": 1,
                  "coordinates": Array [
                    52.369553943508,
                    4.85309666406198,
                  ],
                  "facilities": Array [
                    "Wi-Fi",
                    "Washing machine",
                    "Towels",
                  ],
                  "guests": "Max 2 adults",
                  "id": 2,
                  "image": "img/apartment-02.jpg",
                  "images": Array [
                    "img/room.jpg",
                    "img/apartment-01.jpg",
                    "img/apartment-02.jpg",
                    "img/apartment-03.jpg",
                    "img/studio-01.jpg",
                    "img/apartment-01.jpg",
                  ],
                  "inBookmarks": false,
                  "isPremium": false,
                  "name": "Wood and stone place",
                  "price": 254,
                  "rating": 5,
                  "rooms": "1 Bedrooms",
                  "text": Array [
                    "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.",
                    "An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.",
                  ],
                  "type": "apartment",
                  "zoom": 10,
                },
              ]
            }
            type="main"
            zoom={10}
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;
